File: Package/CS/Client.AE/ObservedNodeCollection.cs
// Copyright (c) Traeger Industry Components GmbH. All Rights Reserved.

namespace AE
{
    using System.Collections.ObjectModel;

    using Opc.UaFx;
    using Opc.UaFx.Client;

    internal class ObservedNodeCollection : KeyedCollection<string, ObservedNode>
    {
        public ObservedNodeCollection()
            : base()
        {
        }

        public ObservedNode Add(OpcClient client, OpcNodeId parentNodeId, string name)
        {
            var item = new ObservedNode(parentNodeId, name);
            item.Initialize(client);

            this.Add(item);
            return item;
        }

        protected override string GetKeyForItem(ObservedNode item)
        {
            return item.Id.ToString(OpcNodeIdFormat.Foundation);
        }
    }
}

File: Package/VB/Client.AE/ObservedNodeCollection.vb
' Copyright (c) Traeger Industry Components GmbH. All Rights Reserved.

Imports System
Imports System.Collections.ObjectModel

Imports Opc.UaFx
Imports Opc.UaFx.Client

Namespace AE
    Friend Class ObservedNodeCollection
        Inherits KeyedCollection(Of String, ObservedNode)

        Public Sub New()
        End Sub

        Public Overloads Function Add(
                client As OpcClient,
                parentNodeId As OpcNodeId,
                name As String) As ObservedNode
            Dim item As New ObservedNode(parentNodeId, name)
            item.Initialize(client)

            Me.Add(item)
            Return item
        End Function

        Protected Overrides Function GetKeyForItem(item As ObservedNode) As String
            Return item.Id.ToString(OpcNodeIdFormat.Foundation)
        End Function
    End Class
End Namespace
